bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2BridgeHandler.java

```diff
@@ -0,0 +1,776 @@
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                                "@text/offline.api2.conf-error.read-only");
+                    } catch (AssetNotLoadedException e) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                "@text/offline.api2.conf-error.assets-not-loaded");
+                    } catch (InterruptedException e) {
+                        return;
+                    }
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "@text/offline.api2.conf-error.not-authorized");
+                }
+                break;
+
+            case COMMUNICATION_ERROR:
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "@text/offline.communication-error");
+                retryConnection = connectRetriesRemaining > 0;
+                break;
+
+            case HANDLER_INITIALIZING_ERROR:
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "@text/offline.api2.conf-error.assets-not-loaded");
+                break;
+
+            case NONE:
+            default:
+                updateSelf(); // go online
+                break;
+        }
+
+        int milliSeconds;
+        if (retryApplicationKey) {
+            // short delay used during attempts to create or validate an application key
+            milliSeconds = FAST_SCHEDULE_MILLI_SECONDS;
+            applKeyRetriesRemaining--;
+        } else {
+            // default delay, set via configuration parameter, used as heart-beat 'just-in-case'
+            Clip2BridgeConfig config = getConfigAs(Clip2BridgeConfig.class);
+            milliSeconds = config.checkMinutes * 60000;
+            if (retryConnection) {
+                // exponential back off delay used during attempts to reconnect
+                int backOffDelay = 60000 * (int) Math.pow(2, RECONNECT_MAX_TRIES - connectRetriesRemaining);
+                milliSeconds = Math.min(milliSeconds, backOffDelay);
+                connectRetriesRemaining--;
+            }
+        }
+
+        // this method schedules itself to be called again in a loop..
+        cancelTask(checkConnectionTask, false);
+        checkConnectionTask = scheduler.schedule(() -> checkConnection(), milliSeconds, TimeUnit.MILLISECONDS);
+    }
+
+    /**
+     * If a child thing has been added, and the bridge is online, update the child's data.
+     */
+    public void childInitialized() {
+        if (thing.getStatus() == ThingStatus.ONLINE) {
+            updateThingsScheduled(5000);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        if (assetsLoaded) {
+            assetsLoaded = false;
+            scheduler.execute(() -> disposeAssets());
```

Comment:
Same question here as in `initialize()` - is this needed?
```suggestion
            disposeAssets();
```

At least I can't spot any network activity or other slow operations in `disposeAssets()`.

Reply:
Ok. Done.